lib/dev_repo.ml

```diff
@@ -8,15 +8,28 @@ let to_string t = t
 
 let repo_name t =
   let uri = Uri.of_string t in
-  let path = Uri.path uri in
+  let path =
+    let possibly_with_trailing_slash = Uri.path uri in
+    String.drop_suffix possibly_with_trailing_slash ~suffix:"/"
+    |> Option.value ~default:possibly_with_trailing_slash
+  in
   let last_path_component =
     match String.rsplit2 ~on:'/' path with
     | None -> path
     | Some (_, last_path_component) -> last_path_component
   in
-  match String.lsplit2 ~on:'.' last_path_component with
-  | None -> last_path_component
-  | Some (repo_name, _ext) -> repo_name
+  let name =
+    match String.lsplit2 ~on:'.' last_path_component with
+    | None -> last_path_component
+    | Some (repo_name, _ext) -> repo_name
+  in
+  if String.equal name "" then
```

Comment:
I'd just match on `name` here, but that's just me.

```ocaml
match name with
| "" -> Rresult.R.error_msgf "unexpected empty string while computing name for dev_repo: %s" t
| name -> Ok name
```

Reply:
I ended up rewriting this whole function after I found some more edge cases that it doesn't handle (multiple trailing slashes, leading periods in the final component of the path)